packages/cli/src/utils/run-all-tests.utils.ts

```diff
@@ -39,3 +56,70 @@ export const getTestsToRun: (options: {
           cached.title === title
       )
   );
+
+  const testCasesMissingBaseReplayId = uncachedTestCases.filter(
+    (testCase) => testCase.baseReplayId == null
+  );
+  const testCasesWithBaseReplayId = uncachedTestCases.flatMap(
+    (testCase): TestCase[] =>
+      // Note: explictly setting the baseReplayId is required for typescript to be happy with types
+      testCase.baseReplayId == null
+        ? []
+        : [{ ...testCase, baseReplayId: testCase.baseReplayId }]
+  );
+
+  if (testCasesMissingBaseReplayId.length === 0) {
+    return testCasesWithBaseReplayId;
+  }
+
+  if (baseCommitSha == null) {
+    const namesOfInvalidTests = testCasesMissingBaseReplayId
+      .map((test) => `"${test.title}"`)
+      .join(", ");
+    throw new Error(
+      `The following test cases are missing a baseReplayId: ${namesOfInvalidTests}.` +
+        " Please either run with the --baseCommitSha option, or add a baseReplayId to all test cases."
+    );
+  }
+
+  const baseReplayIdBySessionId = await getBaseReplayIdsBySessionId({
+    client,
+    baseCommitSha,
+  });
+
+  return uncachedTestCases.flatMap((test) => {
+    if (test.baseReplayId != null) {
+      // Note: explictly setting the baseReplayId is required for typescript to be happy with types
+      return [{ ...test, baseReplayId: test.baseReplayId }];
```

Comment:
```suggestion
      return [{ ...test }];
```

Reply:
Ah good shout! This was actually out of date (I used to have a version of TestCase with the baseReplayId required and one without, and this was needed for converting from the later to the former, but it's no longer needed)